updates such that gateway-rs builds as lib

lthiery opened the pull request:
gateway-rs is not up to date with latest helium-proto and the declaration on "main" is inaccurate

lthiery committed:
lock toml to supported proto/beacon version

lthiery committed:
ci: update upload_artifact version

madninja commented on Cargo.toml:
```diff
@@ -18,7 +18,7 @@ members = ["lorawan"]
 serde = { version = "1", features = ["rc", "derive"] }
 rust_decimal = { version = "1", features = ["serde-with-float"] }
-helium-proto = { git = "https://github.com/helium/proto", branch = "master", features = [
+helium-proto = { git = "https://github.com/helium/proto", rev = "d94ed4b", features = [
```
Sorry what happens on master that breaks things here? I see another fix went in in a different downstream crate to work around things. May be easier to just adjust code?

lthiery replied:
```
error[E0004]: non-exhaustive patterns: `Ok(helium_proto::services::router::envelope_down_v1::Data::PacketAck(_))` not covered
```

lthiery replied:
looks like a new message exists

madninja replied:
we should be able to handle that though..

lthiery replied:
Yea i'm just not sure what to do with it. I'm just trying to get this usable as a lib at the moment

madninja replied:
I believe it was added to HPR as part of https://github.com/helium/gateway-rs/pull/475

madninja replied:
I still believe the PR above should be fixed and merged up or this PR fixed to handle the case

lthiery replied:
I would think be able to build at all would be a good enough PR on its own.

Updating is out of scope of what I can do at the moment

lthiery replied:
I implemented a warning but there is also packet_ack_interval in PacketRouterRegisterV1

madninja replied:
that's ok, as long as it's not set I believe HPR will not issue regular acks

madninja reviewed (commented):
The build issue isn't on the clippy side, it's the cross toolchian not supporting the newer rust version yet that bumps the lock version apparently?

lthiery commented:
> The build issue isn't on the clippy side, it's the cross toolchian not supporting the newer rust version yet that bumps the lock version apparently?

Yea I was seeing if I could bump the toolchain version but too much of a hassle

lthiery committed:
address new clippy lints

lthiery committed:
semtech-udp 0.12 update

lthiery commented:
Shall we just deprecate the quay step and leave its recovery as an open issue?

madninja commented:
> Shall we just deprecate the quay step and leave its recovery as an open issue?

Nope.. looking at it.. perhaps address the other comments?

lthiery commented:
> > Shall we just deprecate the quay step and leave its recovery as an open issue?
> 
> 
> 
> Nope.. looking at it.. perhaps address the other comments?

I addressed all comments and was awaiting response. Things were stalled on quay

lthiery committed:
disable quay from forks

madninja reviewed (changes requested):
Looks like it missed a spot

lthiery committed:
bump helium-proto
